cf/commands/user/space_users.go

```diff
@@ -66,72 +83,94 @@ func (cmd *SpaceUsers) Execute(c flags.FlagContext) {
 	spaceName := c.Args()[1]
 	org := cmd.orgReq.GetOrganization()
 
-	space, apiErr := cmd.spaceRepo.FindByNameInOrg(spaceName, org.Guid)
-	if apiErr != nil {
-		cmd.ui.Failed(apiErr.Error())
+	space, err := cmd.spaceRepo.FindByNameInOrg(spaceName, org.Guid)
+	if err != nil {
+		cmd.ui.Failed(err.Error())
 	}
 
-	cmd.ui.Say(T("Getting users in org {{.TargetOrg}} / space {{.TargetSpace}} as {{.CurrentUser}}",
-		map[string]interface{}{
-			"TargetOrg":   terminal.EntityNameColor(org.Name),
-			"TargetSpace": terminal.EntityNameColor(space.Name),
-			"CurrentUser": terminal.EntityNameColor(cmd.config.Username()),
-		}))
+	printer := cmd.getPrinter()
+	printer.printUsers(org, space, cmd.config.Username())
+}
 
+func (cmd *SpaceUsers) getPrinter() userPrinter {
 	var spaceRoleToDisplayName = map[string]string{
 		models.SPACE_MANAGER:   T("SPACE MANAGER"),
 		models.SPACE_DEVELOPER: T("SPACE DEVELOPER"),
 		models.SPACE_AUDITOR:   T("SPACE AUDITOR"),
 	}
 
-	var usersMap = make(map[string]plugin_models.GetSpaceUsers_Model)
-
-	var users []models.UserFields
-	for _, role := range spaceRoles {
-		displayName := spaceRoleToDisplayName[role]
-
-		if cmd.config.IsMinApiVersion("2.21.0") {
-			users, apiErr = cmd.userRepo.ListUsersInSpaceForRoleWithNoUAA(space.Guid, role)
-		} else {
-			users, apiErr = cmd.userRepo.ListUsersInSpaceForRole(space.Guid, role)
+	if cmd.pluginCall {
+		return &pluginPrinter{
+			pluginModel:            cmd.pluginModel,
+			usersMap:               make(map[string]plugin_models.GetSpaceUsers_Model),
+			userLister:             cmd.getUserLister(),
+			spaceRoleToDisplayName: spaceRoleToDisplayName,
 		}
+	}
+	return &uiPrinter{
+		ui:                     cmd.ui,
+		userLister:             cmd.getUserLister(),
+		spaceRoleToDisplayName: spaceRoleToDisplayName,
+	}
+}
 
-		cmd.ui.Say("")
-		cmd.ui.Say("%s", terminal.HeaderColor(displayName))
+func (cmd *SpaceUsers) getUserLister() func(spaceGuid string, role string) ([]models.UserFields, error) {
+	if cmd.config.IsMinApiVersion("2.21.0") {
+		return cmd.userRepo.ListUsersInSpaceForRoleWithNoUAA
+	}
+	return cmd.userRepo.ListUsersInSpaceForRole
+}
 
+func (p *pluginPrinter) printUsers(_ models.Organization, space models.Space, _ string) {
+	for role, _ := range p.spaceRoleToDisplayName {
+		users, _ := p.userLister(space.Guid, role)
```

Comment:
Mainly because there are no tests that cover that case. I'll add a test before merging this. It's not finished, and will also go through team code review.